syntax = "proto3";


message Observation {
  bytes id_data = 1;
  repeated int32 id_dim = 2;
  string id_dtype = 3;

  bytes obs_data = 4;
  repeated int32 obs_dim = 5;
  string obs_dtype = 6;

  bytes reward_data = 7;
  repeated int32 reward_dim = 8;
  string reward_dtype = 9;

  bytes done_data = 10;
  repeated int32 done_dim = 11;
  string done_dtype = 12;
}

message Action {
  bytes action_data = 13;
  repeated int32 action_dim = 14;
  string action_dtype = 15;

  bytes id_data = 16;
  repeated int32 id_dim = 17;
  string id_dtype = 18;
}

// service
service ModelCall {
    rpc MakeAction(stream Observation) returns (stream Action) {}
}